app/assets/javascripts/discourse/app/components/modal/discard-draft.js

```diff
@@ -0,0 +1,25 @@
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import Component from "@glimmer/component";
+
+export default class DiscardDraftModal extends Component {
+  @service modal;
+  showSaveDraftButton = this.args.model.allowSaveDraft;
```

Comment:
IMO we could just reference `@allowSaveDraft` directly in the template rather than wrapping it with a different name.